standard/statements.md

```diff
@@ -604,6 +605,15 @@ Local function bodies are always reachable. The endpoint of a local function dec
 
 If the type of the argument to a local function is `dynamic`, the function to be called must be resolved at compile time, not runtime.
 
+A local function shall not be used in an expression tree.
+
+A static local function
+
+- May reference static members, type parameters, constant definitions and static local functions from the enclosing scope.
+- Shall not reference `this` or `base` nor instance members from an implicit `this` reference, nor local variables, parameters, or non-static local functions from the enclosing scope. However, all these are permitted in a `nameof()` expression.
+
+Overload resolution of a call within a local function is not affected by whether the local function is static.
```

Comment:
I think this is correct:

```csharp
class Program
{
    public void Method(int x = 10)
    {
    }

    static void Method(string y = "")
    {
    }

    void M()
    {
        StaticLocal();
        NonStaticLocal();
        static void StaticLocal()
        {
            Method();
        }

        void NonStaticLocal()
        {
            Method();
        }
    }
}
```
Both calls to Method are errors.

Reply:
Next attempt:

```csharp
class Program
{
    public void Method(string x)
    {
    }

    static void Method(object x)
    {
    }

    void M()
    {
        StaticLocal();
        NonStaticLocal();
        static void StaticLocal()
        {
            Method("");
        }

        void NonStaticLocal()
        {
            Method("");
        }
    }

    static void StaticM()
    {
        Method("");
    }
}
```

Reply:
Interestingly, if you copy `StaticLocal` into `StaticM` (and call it from `StaticM`) then it's fine.